ddtrace/tracer/span.go

```diff
@@ -106,6 +107,31 @@ func (s *span) BaggageItem(key string) string {
 	return s.context.baggageItem(key)
 }
 
+// LinkSpan sets a casuality link to another span via a spanContext. It also
+// stores details about the link in an attributes map.
+func (s *span) LinkSpan(spanContext ddtrace.SpanContext, attributes map[string]string) {
+	traceIDLower := spanContext.TraceID()
+	var traceIDUpper uint64
+	if w3Cctx, ok := spanContext.(ddtrace.SpanContextW3C); ok {
+		traceIDHex := w3Cctx.TraceID128()[:16]
+		traceIDUpper, _ = strconv.ParseUint(traceIDHex, 16, 64)
+	}
+
+	spanID := spanContext.SpanID()
+	link := ddtrace.SpanLink{}
+	link.TraceID = traceIDLower
+	link.TraceIDHigh = traceIDUpper
+	link.SpanID = spanID
+	link.Attributes = attributes
+	// TODO: Add support for setting tracestate and traceflags
+	s.AddLink(link)
+}
+
+// AddLink sets a mapping between two spans via ddtrace.SpanLink struct
+func (s *span) AddLink(link ddtrace.SpanLink) {
+	s.SpanLinks = append(s.SpanLinks, link)
+}
+
```

Comment:
```suggestion
```

As mentioned above, this should be removed since we can't extend the interface that way.